Neos.ContentRepository.LegacyNodeMigration/Tests/Behavior/Bootstrap/FeatureContext.php

```diff
@@ -123,10 +124,14 @@ public function iRunTheEventMigration(string $contentStream = null): void
     {
         $nodeTypeManager = $this->currentContentRepository->getNodeTypeManager();
         $propertyMapper = $this->getObject(PropertyMapper::class);
-        $contentGraph = $this->currentContentRepository->getContentGraph();
-        $nodeFactory = (new \ReflectionClass($contentGraph))
+        $contentGraphFinder = $this->currentContentRepository->projectionState(\Neos\ContentRepository\Core\ContentGraphFinder::class);
+        // FIXME: Dirty
+        $contentGraphFactory = (new \ReflectionClass($contentGraphFinder))
+            ->getProperty('contentGraphFactory')
+            ->getValue($contentGraphFinder);
+        $nodeFactory = (new \ReflectionClass($contentGraphFactory))
             ->getProperty('nodeFactory')
-            ->getValue($contentGraph);
+            ->getValue($contentGraphFactory);
```

Comment:
Nice reflection and stuff ^^ fyi i started a conversation here about this https://github.com/neos/neos-development-collection/issues/4634

Reply:
This is so bad, but it's just an extension of what was there before, we need to solve this nicely, and if it's just for tests then we should change how we test.

Reply:
We should probably just create some CR Service for the behat tests – that has access to all the `ContentRepositoryServiceFactoryDependencies` including the `propertyConverter` that is the only thing we're extracting here.

But we can certainly solve that in a follow-up!

Reply:
Actually this is much more an issue of the `LegacyMigrationService` in this case.
We can definitely fix that and make it easier to be used from tests!